src/main/java/Duke.java

```diff
@@ -1,10 +1,150 @@
+package ip.src.main.java;
+import java.util.Objects;
+import java.util.Scanner;
+import java.util.Arrays;
+import java.util.ArrayList;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+    private static String file = "ip/src/main/java/duke.txt";
+
+    private static ArrayList<String> letter = new ArrayList<>();
+    private static ArrayList<String> done = new ArrayList<>();
+    private static ArrayList<Task> tasks = new ArrayList<>();
+
+    private static int i=0;
+
+    private static void printFileContents(String filePath) throws FileNotFoundException {
+        File f = new File(filePath); // create a File for the given file path
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+        while (s.hasNext()) {
+            System.out.println(s.nextLine());
+        }
+    }
+
+    private static void writeToFile(String filePath, String textToAdd) throws IOException {
+        FileWriter fw = new FileWriter(filePath, true);
+        fw.write(textToAdd);
+        fw.close();
+    }
+    public static void saveFile(int i) {
+        try {
+            writeToFile(file, i + " | ");
+            writeToFile(file, letter.get(i - 1) + " | ");
+            writeToFile(file, done.get(i - 1) + " | ");
+            writeToFile(file, tasks.get(i - 1).description());
+            writeToFile(file, System.lineSeparator());
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    public static void addTask(Task t, int type) {
+        tasks.add(t);
+        if (type == 1) letter.add("D");
+        else if (type == 2) letter.add("E");
+        else if (type == 3) letter.add("T");
```

Comment:
Avoid the use of magic literals. Using a suitable variable name for the constant numbers (which describes the value) would help the understanding of the program. 

Also consider putting the repeated logic into another method.